lib/index.js

```diff
@@ -2,52 +2,52 @@
  * @fileoverview Package exports for @eslint/eslintrc
  * @author Nicholas C. Zakas
  */
-"use strict";
-
 //------------------------------------------------------------------------------
 // Requirements
 //------------------------------------------------------------------------------
 
-const {
+import {
     ConfigArrayFactory,
-    createContext: createConfigArrayFactoryContext
-} = require("./config-array-factory");
-
-const { CascadingConfigArrayFactory } = require("./cascading-config-array-factory");
-const ModuleResolver = require("./shared/relative-module-resolver");
-const { ConfigArray, getUsedExtractedConfigs } = require("./config-array");
-const { ConfigDependency } = require("./config-array/config-dependency");
-const { ExtractedConfig } = require("./config-array/extracted-config");
-const { IgnorePattern } = require("./config-array/ignore-pattern");
-const { OverrideTester } = require("./config-array/override-tester");
-const ConfigOps = require("./shared/config-ops");
-const ConfigValidator = require("./shared/config-validator");
-const naming = require("./shared/naming");
-const { FlatCompat } = require("./flat-compat");
+    createContext as createConfigArrayFactoryContext
+} from "./config-array-factory.js";
+
+import { CascadingConfigArrayFactory } from "./cascading-config-array-factory.js";
+import * as ModuleResolver from "./shared/relative-module-resolver.js";
+import { ConfigArray, getUsedExtractedConfigs } from "./config-array/index.js";
+import { ConfigDependency } from "./config-array/config-dependency.js";
+import { ExtractedConfig } from "./config-array/extracted-config.js";
+import { IgnorePattern } from "./config-array/ignore-pattern.js";
+import { OverrideTester } from "./config-array/override-tester.js";
+import * as ConfigOps from "./shared/config-ops.js";
+import ConfigValidator from "./shared/config-validator.js";
+import * as naming from "./shared/naming.js";
+import { FlatCompat } from "./flat-compat.js";
 
 //-----------------------------------------------------------------------------
 // Exports
 //-----------------------------------------------------------------------------
 
-module.exports = {
-
-    Legacy: {
-        ConfigArray,
-        createConfigArrayFactoryContext,
-        CascadingConfigArrayFactory,
-        ConfigArrayFactory,
-        ConfigDependency,
-        ExtractedConfig,
-        IgnorePattern,
-        OverrideTester,
-        getUsedExtractedConfigs,
-
-        // shared
-        ConfigOps,
-        ConfigValidator,
-        ModuleResolver,
-        naming
-    },
+const Legacy = {
+    ConfigArray,
+    createConfigArrayFactoryContext,
+    CascadingConfigArrayFactory,
+    ConfigArrayFactory,
+    ConfigDependency,
+    ExtractedConfig,
+    IgnorePattern,
+    OverrideTester,
+    getUsedExtractedConfigs,
+
+    // shared
+    ConfigOps,
+    ConfigValidator,
+    ModuleResolver,
+    naming
+};
```

Comment:
It seems we should add environments here since ESLint needs that:

https://github.com/eslint/eslint/blob/e31f49206f94e2b3977ec37892d4b87ab1e46872/lib/linter/linter.js#L19

The exported name could be `BuiltInEnvironments`?

Reply:
I'll add that.